lib/seattleflu/db/cli/command/kit_result.py

```diff
@@ -0,0 +1,206 @@
+    """
+    Parse self-test kit results from manifest using columns listed in <config.yaml>
+
+    <config.yaml> must be a file with one YAML document in it.
+    Example document:
+
+    \b
+        ---
+        workbook: OneDrive/Self-Test Samples 2018-2019.xlsx
+        sheet: Sheet1
+        columns:
+          kit_barcode: "Box Barcode ID"
+          test_date: "Cepheid Test date"
+          strip_barcode: "Strip Barcode"
+          strip_flu_a: "Strip Flu A"
+          strip_flu_b: "Strip Flu B"
+          rdt_barcode: "RDT Barcode"
+          rdt_flu_a: "RDT Flu A"
+          rdt_flu_b: "RDT Flu B"
+          rdt_rsv: "RDT RSV"
+          utm_barcode: "UTM Sample Barcode "
+          utm_flu_a: "UTM Flu A"
+          utm_flu_b: "UTM Flu B"
+          utm_rsv: "UTM RSV"
+
+    Kit result records are output to stdout as newline-delimited JSON records.
+    You will likely want to redirect stdout to a file.
+    """
+    config = yaml.safe_load(config_file)
+
+    try:
+        kit_result = {
+            "workbook": config["workbook"],
+            "sheet": config["sheet"],
+            "kit_barcode": config["columns"]["kit_barcode"],
+            "test_date": config["columns"]["test_date"],
+            "strip_barcode": config["columns"]["strip_barcode"],
+            "strip_flu_a": config["columns"]["strip_flu_a"],
+            "strip_flu_b": config["columns"]["strip_flu_b"],
+            "rdt_barcode": config["columns"]["rdt_barcode"],
+            "rdt_flu_a": config["columns"]["rdt_flu_a"],
+            "rdt_flu_b": config["columns"]["rdt_flu_b"],
+            "rdt_rsv": config["columns"]["rdt_rsv"],
+            "utm_barcode": config["columns"]["utm_barcode"],
+            "utm_flu_a": config["columns"]["utm_flu_a"],
+            "utm_flu_b": config["columns"]["utm_flu_b"],
+            "utm_rsv": config["columns"]["utm_rsv"]
+        }
+    except KeyError as key:
+        LOG.error(f"Required key «{key}» missing from config")
+        raise key from None
+
+    barcode_columns = [v for k,v in kit_result.items() if "barcode" in k]
+
+    manifest = load_manifest_data(
+        workbook = kit_result["workbook"],
+        sheet = kit_result["sheet"],
+        barcode_columns = barcode_columns)
+
+    strip = subset_manifest(manifest, "strip", kit_result)
+
+    rdt = subset_manifest(manifest, "rdt", kit_result)
+
+    utm = subset_manifest(manifest, "utm", kit_result)
+
+    for i in manifest.index:
+        print(as_json(strip.loc[i].to_dict()))
+        print(as_json(rdt.loc[i].to_dict()))
+        print(as_json(utm.loc[i].to_dict()))
+
+
+def load_manifest_data(workbook: str, sheet:str, barcode_columns: list) -> tuple:
+    """
+    Load the manifest given the *workbook* and *sheet*.
+    Will drop any records that have NA or duplicated values for the given *barcode_columns*.
+    """
+    replacements = {
+        pd.np.nan: None,
+        "": None,
+        "NA": None,
+        "N/A": None,
+        "?": None
+    }
+
```

Comment:
I'm curious here: why did you want to convert pandas `NaN` to `None`?

Reply:
I think `NaN` gets converted to a string "NaN" at some point and would not be dropped with the other null values.